src/tz_ohos.rs

```diff
@@ -0,0 +1,48 @@
+//! OpenHarmony doesn't have `/etc/localtime`, we have to use it's "Time Service" to get the timezone information:
+//!
+//! - [API Refence](https://gitee.com/openharmony/docs/blob/43726785b4033887cd1a838aaaca5e255897a71e/en/application-dev/reference/apis-basic-services-kit/_time_service.md#oh_timeservice_gettimezone)
+
+use crate::GetTimezoneError;
+use std::ffi::CStr;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[repr(C)]
+#[allow(non_camel_case_types)]
+#[allow(dead_code)]
+enum TimeService_ErrCode {
+    TIMESERVICE_ERR_OK = 0,
+    TIMESERVICE_ERR_INTERNAL_ERROR = 13000001,
+    TIMESERVICE_ERR_INVALID_PARAMETER = 13000002,
+}
+
+#[link(name = "time_service_ndk", kind = "dylib")]
+extern "C" {
+    fn OH_TimeService_GetTimeZone(timeZone: *mut std::ffi::c_char, len: u32)
+        -> TimeService_ErrCode;
+}
+
+/// TODO: Change this `CStr::from_bytes_until_nul` once MSRV was bumped above 1.72.0
```

Comment:
```suggestion
/// TODO: Change this `CStr::from_bytes_until_nul` once MSRV was bumped above 1.69.0
```

per the docs: https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.from_bytes_until_nul

Reply:
Ah it's looks like [from_bytes_with_nul](https://doc.rust-lang.org/std/ffi/struct.CStr.html#method.from_bytes_with_nul)